controlplane/rosa/api/v1beta2/rosacontrolplane_types.go

```diff
@@ -76,6 +74,38 @@ type RosaControlPlaneSpec struct { //nolint: maligned
 	// - ocmApiUrl: Optional, defaults to 'https://api.openshift.com'
 	// +optional
 	CredentialsSecretRef *corev1.LocalObjectReference `json:"credentialsSecretRef,omitempty"`
+
+	// Network config for the ROSA-HCP cluster.
+	// +optional
+	Network *NetworkSpec `json:"network,omitempty"`
+
+	// The compute machine type to use, for example `r5.xlarge`. Machine type ref; https://aws.amazon.com/ec2/instance-types/
+	// +optional
+	ComputeMachineType string `json:"computeMachineType,omitempty"`
+
+	// Autoscaling specifies auto scaling behaviour for this NodePool.
+	// +optional
+	Autoscaling *expinfrav1.RosaMachinePoolAutoScaling `json:"autoscaling,omitempty"`
+}
+
+type NetworkSpec struct {
+	// IP addresses block used by OpenShift while installing the cluster, for example "10.0.0.0/16".
+	// +kubebuilder:validation:Format=cidr
+	MachineCIDR *string `json:"machineCIDR"`
+
+	// IP address block from which to assign pod IP addresses, for example `10.128.0.0/14`.
+	// +kubebuilder:validation:Format=cidr
+	// +optional
+	PodCIDR string `json:"podCIDR,omitempty"`
+
+	// IP address block from which to assign service IP addresses, for example `172.30.0.0/16`.
+	// +kubebuilder:validation:Format=cidr
+	// +optional
+	ServiceCIDR string `json:"serviceCIDR,omitempty"`
+
+	// Network host prefix which is defaulted to `23` if not specified.
```

Comment:
Please add the default to the kubebuilder configuration (and can we validate this field?)